tests/test_other.py

```diff
@@ -10233,6 +10233,63 @@ def test_split_main_module(self):
     self.assertIn('Hello from main!', result)
     self.assertIn('Hello from lib!', result)
 
+  def test_split_module_ex(self):
+    initialTableSize = 5632
+    self.set_setting('SPLIT_MODULE')
+    self.set_setting('WASM', 1)
+    self.set_setting('WASM_BIGINT')
+    self.set_setting('EVAL_CTORS', 0)
+    self.set_setting('INLINING_LIMIT', 1)
+    self.set_setting('FORCE_FILESYSTEM', 1)
+    self.set_setting('LEGACY_GL_EMULATION', 0)
+    self.set_setting('DISABLE_EXCEPTION_CATCHING', 0)
+    self.set_setting('ALLOW_MEMORY_GROWTH', 1)
+    self.set_setting('NO_EXIT_RUNTIME', 0)
+    self.set_setting('GL_TRACK_ERRORS', 0)
+    self.set_setting('WASM_OBJECT_FILES', 0)
+    self.set_setting('INCLUDE_FULL_LIBRARY', 1) # https://github.com/emscripten-core/emscripten/issues/13794
+    self.emcc_args += ['-sMAIN_MODULE=2']
+    # Exception: FROZEN_CACHE is set, but cache file is missing: sysroot/lib/wasm32-emscripten/lto-pic/libgl-webgl2-full_es3.a
+    # self.emcc_args += ['-sUSE_WEBGL2']
+    # self.emcc_args += ['-sUSE_GLFW=3']
+    # self.emcc_args += ['-sFULL_ES3=1']
+
+    self.emcc_args += ['-Wno-experimental']
+    self.emcc_args += ['-std=c++17']
+    self.emcc_args += ['-D_LIBCPP_ENABLE_CXX17_REMOVED_AUTO_PTR']
```

Comment:
What are these two lines needed for?

Reply:
But what has c++17 or auto-pointer got to do with SPLIT_MODULE?

Reply:
Maybe we can write a separate standalone test for `_LIBCPP_ENABLE_CXX17_REMOVED_AUTO_PTR`?

Reply:
I understand that me pushing back on this change landing is probably frustrating for you so I'm happy to write that standalone test.. which should allow us to remove at least that part from this SPLIT_MODULE test.

Reply:
Can you also remove `-std=c++17 -D_LIBCPP_ENABLE_CXX17_REMOVED_AUTO_PTR -Wno-deprecated-declarations` here now too?